### PR TITLE
Rename package to sqlalchemy_redshift

### LINKED ISSUE #40: Migration plan for moving to the sqlalchemy-redshift org
There are some details we'll need to deal with when this project moves to the https://github.com/sqlalchemy-redshift org.

* Documentation
  * [ ] Make sure we're using sqlachemy-redshift and sqlalchemy_redshift consistently everywhere
  * [x] Make sure any references to the repo are correct
* [Read the Docs](https://readthedocs.org/projects/redshift-sqlalchemy)
  * [x] Change the url of docs to `sqlalchemy-redshift` (it's currently inverted)
  * [x] Set up a webhook on the repo so docs will build automatically.
* PyPI
  * [x] [Make redshift-sqlalchemy rely on sqlalchemy-redshift](https://github.com/binarydud/redshift_sqlalchemy/pull/33)

I wanted to make sure to document thoughts as I have them. Other things we'll need to keep in mind?

### PR DESCRIPTION
**Pull request by jklukas**
Fixes #40.

### CODE REVIEW

File: redshift_sqlalchemy/__init__.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-from sqlalchemy.dialects import registry
+import sqlalchemy_redshift
 
-registry.register("redshift", "redshift_sqlalchemy.dialect", "RedshiftDialect")
-registry.register(
-    "redshift+psycopg2", "redshift_sqlalchemy.dialect", "RedshiftDialect"
-)
+
+# All references to module redshift_sqlalchemy will map to sqlalchemy_redshift
+sys.modules['redshift_sqlalchemy'] = sqlalchemy_redshift
```
Comment (jklukas) [inline redshift_sqlalchemy/__init__.py]: I just discovered `sys.modules`. This appears to do the magic that we want. It's possible to do imports like `from redshift_sqlalchemy.dialect import RedshiftDDLCompiler`.

Comment (graingert) [inline redshift_sqlalchemy/__init__.py]: this is correct but it breaks:

```
from redshift_sqlalchemy import dialect
reload(dialect)
```

**Comment by graingert** [comment]
can you split this into two commits and submit them one at a time so travis sees each one:

1. where you
   * rename the module redshift_sqlalchemy
   * create the compat module
   * update the changelog
2. where you update the tests to point at sqlalchemy_redshift

This way we can prove the compatability module still works

File: redshift_sqlalchemy/__init__.py [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
-)
+
+DEPRECATION_MESSAGE = """\
+redshift_sqlalchemy has been renamed to sqlalchemy_redshift.
+
+The redshift_sqlalchemy compatibility package will be removed in
+a future release, so it is recommended to update all package references.
+"""
+
+warnings.warn(DEPRECATION_MESSAGE, DeprecationWarning)
```
Comment (jklukas) [inline redshift_sqlalchemy/__init__.py]: `DeprecationWarning` is ignored by default, but would seem to be the semantically correct choice here.

**Comment by jklukas** [comment]
I understand the philosophy of why you'd want to split this up, but I'm not inclined to put in the extra work to make it happen.

File: redshift_sqlalchemy/__init__.py (hunk 2 above)
Comment (graingert) [inline redshift_sqlalchemy/__init__.py]: nice

**Comment by jklukas** [comment]
Rebased to edit the commit with broken README.rst.

File: CHANGES.rst [hunk 3]
```diff
@@ -1,7 +1,10 @@
 0.3.2 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Change the name of the package to `sqlalchemy_redshift` to match the naming
+  convention for other dialects; the `redshift_sqlalchemy` now references
```
Comment (graingert) [inline CHANGES.rst]: missed a word? Should this be "the deprecated `redshift_sqlalchemy` now references..."

**Comment by jklukas** [comment]
Updated the CHANGELOG message.

**Comment by graingert** [comment]
Perfect! Merge when ready

**Pull request merged**